### DISCUSSION TITLE
PCB capacitor

### DISCUSSION THREAD

**Discussion by 2Kasi2** [RonMcKay/capacitive-soil-moisture-sensor#12] [Q&A]
How did you design the shape of the PCB capacitor and what is its capacitance? I would like to design my own version but cant seem to find a way to calculate the capacitance of planar capacitor, such as the one you used.

**Comment by adfx** [comment]
For what reason do you need to know exact values? It's as simple as just measuring two values. One in very moist conditions the other very dry. That's it. Hope that helps.

Regards eddy

**Reply by 2Kasi2** [reply]
I want to make my own, if the capacitance value starts too low (in air, lets say less than 1pF) the change in voltage will be too low for the MCU to notice. Is the value is too high (in air, for example >30pF) the highest voltage value will be low and the sensor will lose resolution. See the picture attached for explenation.
![delete](https://user-images.githubusercontent.com/67429462/202518590-4c5cf4ec-a245-433c-9920-9c046d4b57ff.png)
Also it would be nice to know how the shape affects the change in capacitance from calculations...

Source for the picture: https://wemakethings.net/2012/09/26/capacitance_measurement/

**Reply by RonMcKay** [reply]
I was also thinking about how to increase the value range of the sensor.
In the formula you are showing, we can also adjust $t$ by changing the frequency of the Arduino PWM (this could be found empirically in a loop in a kind of 'calibration mode' - I might be working on this one in the future) and we might be able to use different resistor values $R$.
So if $R$, $C$ and $V_S$ are fixed, we can compute the optimal frequency $t^*$ as follows:

$$t^* = arg\ max_{t\in T} \left[V_S\cdot\left(1-\exp^{-\frac{t}{RC_l}}\right) - V_S\cdot\left(1-\exp^{-\frac{t}{RC_u}}\right)\right] = arg\ max_{t\in T} \left[\exp^{-\frac{t}{RC_u}} - \exp^{-\frac{t}{RC_l}}\right]$$

Here, $C_l$ and $C_u$ are lower and upper bound of our PCB capacitor and $T$ is an appropriate value range for the frequencies. In practice, we could approximate $t^*$ by probing a few points $T'\subset T$ while the sensor is submerged in water and in air.

I am not sure how much of a difference this will make for the final value range of the sensor, but might be interesting to try.

**Reply by RonMcKay** [reply]
Never mind. Just calculated it through and you always get the highest sensor value range for the highest frequency that you can achieve with your hardware. However, higher frequency also means that the curve from your screenshot will be more bend, resulting in a non-linear sensor behavior (more sensitive for smaller capacitance). So I doubt that my suggested calibration mode would do any good.

**Reply by RonMcKay** [reply]
You can find a paper on the design of planar capacitive sensors here:
[Planar capacitive sensors - designs and applications](https://www.emerald.com/insight/content/doi/10.1108/02602281011010772/full/html?utm_campaign=Emerald_Engineering_PPV_Dec22_RoN)
Therein is also an analysis of different sensor forms and their sensitivity (see screenshot below). But I guess we can conclude, that optimizing the shape of the planar capacitor is not an easy task and more or less an open research question (though I did not do an extensive amount of literature research and would be happy about more recent publications on the topic).

<details>
<summary> Figure 13 from "Planar capacitive sensors - designs and applications" by Hu et al. </summary>
<img src="https://github.com/RonMcKay/capacitive-soil-moisture-sensor/assets/36410367/22f76b2d-9e34-4a46-9b26-93d83b7fff9d">
</details>

**Comment by RonMcKay** [comment]
Hi @2Kasi2,

actually, I did not calculate or optimize the shape of the PCB capacitor in any way. I based my design on already existing ones and shaped it according to my desired total PCB width (so that it exactly fits 2x AAA batteries).

If you find some resources regarding optimizing the shape and/or size, I would also be interested in this. :smile:
